README.md

```diff
@@ -1,8 +1,35 @@
 # Nickel-nix
 
-An experimental Nix toolkit to use Nickel as a language for writing Nix
+An experimental Nix toolkit to use
+[Nickel](https://github.com/tweag/nickel) as a language for writing nix
 packages, shells and more.
 
+**Caution**: `nickel-nix` is not a full-fledged Nix integration. It is
+currently missing several basic features. `nickel-nix` is intended to be a
+proof-of-concept for writing a Nix derivation using Nickel at the moment,
+without requiring new features either in Nix or in Nickel. The future
+of the integration of Nix and Nickel will most probably involve deeper
+changes to the two projects, and hopefully lead to a more featureful
+and ergonomic solution.
+
+# How to use
+
+We currently provide a few basics development environment for some languages.
+
+You need Nix with flakes enabled to use Nickel-nix:
+
+```shell
+$ mkdir my_project
+$ cd my_project
+$ nix flake init -t github:nickel-lang/nickel-nix#rust-devshell
+```
+
+You can list available development shell templates by running:
+
+```shell
+nix flake show github:nickel-lang/nickel-nix
+```
```

Comment:
# How to use

We currently provide a few basics development environment for some languages.
You need Nix with flakes enabled to use Nickel-nix.

You can list available development shell templates by running:

```shell
nix flake show github:nickel-lang/nickel-nix
```

The following example copies the rust devshell and enters the development environment.

```shell
$ mkdir my_project
$ cd my_project
$ nix flake init -t github:nickel-lang/nickel-nix#rust-devshell
$ nix develop --impure
```

Reply:
Suggestion^^^
Using ``` suggestion breaks with the fenced code blocks so I just added it as a comment.

Reply:
is it right now?